### DISCUSSION TITLE
Layout Gruppieren

### DISCUSSION THREAD

**Discussion by Sineos** [Zefau/ioBroker.jarvis#2049] [Q&A]
Ich stelle diverse Verbraucher als Grafana Graph in einem iframe dar:

![image](https://user-images.githubusercontent.com/4459312/201520102-9a5fbe9e-fed2-4997-bda5-2724e444b340.png)

Mein Problem ist nun, dass das auf meinem Monitor cool ausschaut, aber auf einem Mobil-Gerät zerpflückt wird. Gibt es eine Möglichkeit die Widgets zu gruppieren, dass er quasi die gesamte Gruppe responsive skaliert aber die Anordnung beibehält?

**Comment by mcuiobroker** [comment]
https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/styles/grid

**Reply by Sineos** [reply]
Danke, aber ich habs so nicht hin bekommen. Wahrscheinlich mein Fehler.

Unter `STYLES` hab ich jetzt Folgendes stehen:
```
/* Einstellungen für width <=420 px */

@media all and (max-width:420px) {
	.jarvis-page-4e7b546d-4b83-42f3-a566-4d8dbf9d3e3b .container.desktop > .box:nth-child(3) {
		grid-column: 4 / 7 !important;
		grid-row: 3 / 4 !important;
	}

	.jarvis-page-4e7b546d-4b83-42f3-a566-4d8dbf9d3e3b .container.desktop > .box:nth-child(3) .iFrame {
		height: 100px !important;
	}

	.jarvis-page-4e7b546d-4b83-42f3-a566-4d8dbf9d3e3b .container.desktop > .box:nth-child(4) {
		grid-column: 7 /10 !important;
		grid-row: 3 / 4 !important;
	}

	.jarvis-page-4e7b546d-4b83-42f3-a566-4d8dbf9d3e3b .container.desktop > .box:nth-child(4) .iFrame {
		height: 100px !important;
	}

	.jarvis-page-4e7b546d-4b83-42f3-a566-4d8dbf9d3e3b .container.desktop > .box:nth-child(6) {
		grid-column: 1 /4 !important;
		grid-row: 5 / 6 !important;
	}

	.jarvis-page-4e7b546d-4b83-42f3-a566-4d8dbf9d3e3b .container.desktop > .box:nth-child(6) .iFrame {
		height: 100px !important;
	}

	.jarvis-page-4e7b546d-4b83-42f3-a566-4d8dbf9d3e3b .container.desktop > .box:nth-child(7) {
		grid-column: 4 /7 !important;
		grid-row: 5 / 6 !important;
	}

	.jarvis-page-4e7b546d-4b83-42f3-a566-4d8dbf9d3e3b .container.desktop > .box:nth-child(7) .iFrame {
		height: 100px !important;
	}

	.jarvis-page-4e7b546d-4b83-42f3-a566-4d8dbf9d3e3b .container.desktop > .box:nth-child(8) {
		grid-column: 7 /10 !important;
		grid-row: 5 / 6 !important;
	}

	.jarvis-page-4e7b546d-4b83-42f3-a566-4d8dbf9d3e3b .container.desktop > .box:nth-child(8) .iFrame {
		height: 100px !important;
	}

	.jarvis-page-4e7b546d-4b83-42f3-a566-4d8dbf9d3e3b .container.desktop > .box:nth-child(9) {
		grid-column: 10 /13 !important;
		grid-row: 5 / 6 !important;
	}

	.jarvis-page-4e7b546d-4b83-42f3-a566-4d8dbf9d3e3b .container.desktop > .box:nth-child(9) .iFrame {
		height: 100px !important;
	}
}
```

wobei `4e7b546d-4b83-42f3-a566-4d8dbf9d3e3b` die ID des Tabs ist.
Und angeordnet hab ich die Elemente so:

![image](https://user-images.githubusercontent.com/4459312/203098985-83a6a3fb-e880-4f6f-a6ce-89845226fb54.png)

Edit:
Ich hab keine Ahnung was ich hier tue und nur versucht der Anleitung zu folgen. HTML und speziell CSS gehören für mich in den Bereich der schwarzen Magie.

**Reply by mcuiobroker** [reply]
welche breite hat das Handy? px?
https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/system-effektprog./bildschirmgroesse-handy-jarvis
```
@media all and (max-width:420px) {
	.jarvis-tab-4e7b546d-4b83-42f3-a566-4d8dbf9d3e3b .container.desktop > .box:nth-child(7) {
		grid-column: 1 / 4 !important;
		grid-row: 6 / 10 !important;
	}

	

	.jarvis-tab-4e7b546d-4b83-42f3-a566-4d8dbf9d3e3b .container.desktop > .box:nth-child(8) {
		grid-column: 1 / 4 !important;
		grid-row: 10 / 14 !important;
	}

	

	.jarvis-tab-4e7b546d-4b83-42f3-a566-4d8dbf9d3e3b .container.desktop > .box:nth-child(9) {
		grid-column: 4 /13 !important;
		grid-row: 6 / 14 !important;
	}
}
```
Für welche Ansicht soll das sein? Hoch- oder Quer? Es beginnt jetzt in der 6.ten Zeile evtl erhöhen?
Wenn es so geklappt hat, bitte als Bild vom Handy posten, dann kann ich das in die Doku mit aufnehmen.

**Reply by Sineos** [reply]
Vielen Dank, das geht schon mal in die richtige Richtung.

iPhone7 quer:
![IMG_0047](https://user-images.githubusercontent.com/4459312/203157538-d78f987e-e173-4ce3-aa95-bde9ee695984.PNG)

iPhone7 hoch:
![IMG_0048](https://user-images.githubusercontent.com/4459312/203157540-33bf9c12-8d10-422d-afdc-886c08a7bd99.PNG)

Wäre das ein Feature Request? Weil irgendwie noch Modell- und orientierungsabhängig zu sein ist ja auch keine Lösung

**Reply by mcuiobroker** [reply]
Hab es oben nochmal geändert, bitte nochmal anpassen.
Einstellung ist nur für Hochformat. Für Quer musst du mir die Breite im Querformat geben.
Feature Request kannst du machen.

**Comment by Sineos** [comment]
![IMG_0049](https://user-images.githubusercontent.com/4459312/203159932-01975529-ae1f-40cc-9064-7e5d8a2dac05.PNG)

Das CSS stark in Dir ist mein junger Padawan :wink:

**Reply by mcuiobroker** [reply]
Na das sieht doch gut aus.
